global/scripts/RSNregression.m

```diff
@@ -178,6 +178,49 @@ function RSNregression(InputFile, InputVNFile, Method, ParamsFile, OutputBeta, v
             
             betaICAone = weightedDualRegression(GroupMapcii.cdata, inputConcat, AreaWeights .* ScaledAlignmentQuality);
             [betaICA, NODEts] = weightedDualRegression(normalise(betaICAone), inputConcat, AreaWeights);
+            
+            if strcmp(Method, "tICA_weighted") % based on tICA/scripts/ComputeGroupTICA.m but for one subject as a group
+                NODEts=NODEts';
+                disp(['NODEts'])
+                size(NODEts)
+                SpectraArray = textscan(optional.SpectraParams, '%s', 'Delimiter', {'@'});
+                nTPsForSpectra =str2double(SpectraArray{1}{1});
+                thisStart = 1;
+                TCSRunVarSub = [];
+                for i = 1:size(inputArray, 1)
+                    dtseriesName=[inputArray{i}];
+                    if exist(dtseriesName, 'file')
+                        runLengthStr = my_system(['wb_command -file-information -only-number-of-maps ' dtseriesName]);
+                        disp(['runLengthStr ' runLengthStr])
+                        runLength = str2double(runLengthStr);
+                        nextStart = thisStart + runLength;
+                        TCSRunVarSub = [TCSRunVarSub repmat(std(NODEts(:, thisStart:(nextStart - 1)), [], 2), 1, runLength)];
+    
+                        thisStart = nextStart;
+                    end
+                end
+                
+                sICAtcsvars = std(NODEts, [], 2);
+                NODEts = (NODEts ./ TCSRunVarSub) .* repmat(sICAtcsvars, 1, size(TCSRunVarSub, 2)); %Making all runs contribute equally improves tICA decompositions
+                NODEts(~isfinite(NODEts)) = 0;
+
+                A = load(optional.tICAMM);
+
+                if size(A,1) ~= size(NODEts,1)
+                    error('Mixing matrix to be used does not match dimensions of the sICA components');
+                end
+                
+                % unmix the sICA timeseries
+                W = pinv(A);
+                normicasig = W * NODEts;
+                icasig = normicasig .* repmat(std(A ./ repmat(sICAtcsvars, 1, size(A, 2)))', 1, size(NODEts, 2)); %Unormalize the icasig assuming sICAtcs with std = 1 (approximately undo the original variance normalization)
```

Comment:
It would be nice if we could get closure on whether this is doing something reasonable before merging this.

Reply:
@glasserm might have his explanation on this.

Reply:
I think I put this in an e-mail somewhere. The point is to get the timeseries to have the amplitudes rather than std=1.

Reply:
It appears to be taking the standard deviation of the mixing matrix divided by sICA standard deviations.  I don't follow how that implements what you said.

Reply:
If I recall correctly, the goal was to have the tICA component timeseries variances be not one but have the result reflect what it would have been if the sICA component timeseries variances were one.

Reply:
If the timeseries in question are orthogonal, then you can combine their stdevs by root mean square without reconstructing the actual timeseries.  This does not appear to be doing that (stdev calculates and subtracts the mean first, though it is otherwise root mean square, so perhaps this is why its answer may look reasonable).

Reply:
Also, `A` is the matrix that turns tICA components back into sICA components, right?  That doesn't seem like the correct matrix to use in combination with sICA stdevs.

Reply:
I may be starting to puzzle out the intent, ICA outputs `A` and `normicasig`, such that `A * normicasig` equals the input, but `normicasig` has rows with...probably stdev equal to 1.  If we think of it in terms of `[u s v] = svd(...);`, we want `s * v'`, but ICA gives us something like `u * s` and `v'` instead.  In the case of SVD, because `s` is diagonal and `u` is orthonormal, we can recover the diagonal of `s` from `u * s` via `sqrt(sum((u * s) .^ 2))`.  `A` has the amplitudes corresponding to the tICA of `NODEts` (after per-run normalization and mostly-restoration of original stdev (missing runs stay 0, throwing off the per-run normalization's outcome a bit)), and we...want to pretend `NODEts` had been normalized before ICA?  To stdev=1?  We would have:

input = A * normicasig
pretendinput = diag(1 / stdev(input)) * A * normicasig
...assume normicasig doesn't change...
pretendA = diag(1 / stdev(input)) * A = A ./ repmat(stdev(input), ...)

which starts to look familiar.  If ica spits out timeseries with stdev=1, it would almost work out if the mixing matrix is orthonormal...it probably isn't, and the demeaning part of the stdev is likely undesired on A, but it may be close enough to not be worth the trouble of improving it.  Some of this should get written down (here and in tICA) so we don't have to guess next time.

Reply:
Someone please double check that fastica and icasso spit out stdev = 1 timeseries, and see if you agree with the explanation I added.

Reply:
They are.  Explanation is fine.  The purpose of this is to help reorder to components from strongest to weakest according to the tICA decomposition alone.